backups/files/dump-database.sh

```diff
@@ -0,0 +1,214 @@
+#!/bin/bash
+# shellcheck enable=avoid-nullary-conditions
+# shellcheck enable=check-set-e-suppressed
+
+set -euo pipefail
+
+#####
+# Arguments
+
+# This script expects a single argument, identifier, which corresponds to a
+# number of values in this and other scripts:
+#
+# 1. It names the configuration file within /opt/backups/lib.
+# 2. It names the subdirectory within the backup root (e.g., "/var/lib/backups/$identifier")
+# 3. It names the lockfile and log tag.
+readonly identifier="${1:?USAGE: $0 <identifier>}"
```

Comment:
Can you explain how this assignment works?

Reply:
Ahh... does it just print "Usage: dump-database.sh <identifier>" if the script is executed without an argument?

Reply:
It's shorthand to basically fail on an unset/empty variable, so I'm using it to fail immediately if the only required input is missing.